src/zenml/integrations/lightning/orchestrators/lightning_orchestrator.py

```diff
@@ -103,20 +103,34 @@ def _set_lightning_env_vars(
 
         Args:
             deployment: The pipeline deployment to prepare or run.
+
+        Raises:
+            ValueError: If the user id and api key or username and organization
         """
         settings = cast(
             LightningOrchestratorSettings, self.get_settings(deployment)
         )
-        if settings.user_id:
+        if not settings.user_id or not settings.api_key:
```

Comment:
This condition fails if one of the two is missing, but the error message specifies that only one of them must be set. What is correct?